--- libultrahdr-1.5.1/CMakeLists.txt.orig	2026-09-11 21:02:10.235321744 +0200
+++ libultrahdr-1.5.1/CMakeLists.txt	2026-09-13 10:44:52.046503523 +0200
@@ -230,6 +230,9 @@ endif()
 if(UHDR_WRITE_ISO)
   add_compile_options(-DUHDR_WRITE_ISO)
 endif()
+if(UHDR_ENABLE_SMPTE2094_50)
+  add_compile_options(-DUHDR_ENABLE_SMPTE2094_50)
+endif()
 
 include(CheckCXXCompilerFlag)
 function(CheckCompilerOption opt res)
@@ -378,24 +381,6 @@ find_package(Threads REQUIRED)
 
 include(ExternalProject)
 
-if(UHDR_ENABLE_SMPTE2094_50)
-  if(UHDR_BUILD_DEPS)
-    add_compile_options(-DUHDR_ENABLE_SMPTE2094_50)
-    include(FetchContent)
-    FetchContent_Declare(
-      libsmpte2094_50
-      GIT_REPOSITORY https://github.com/webmproject/libsmpte2094-50.git
-      GIT_TAG v0.1.4)
-    set(OLD_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
-    set(BUILD_SHARED_LIBS OFF)
-    FetchContent_MakeAvailable(libsmpte2094_50)
-    set(BUILD_SHARED_LIBS ${OLD_BUILD_SHARED_LIBS})
-  else()
-    message(WARNING "libsmpte2094-50 package not available")
-    set(UHDR_ENABLE_SMPTE2094_50 OFF)
-  endif()
-endif()
-
 get_directory_property(UHDR_COMPILE_FLAGS COMPILE_OPTIONS)
 string (REPLACE ";" " " UHDR_COMPILE_FLAGS_STR "${UHDR_COMPILE_FLAGS}")
 set(UHDR_CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${UHDR_COMPILE_FLAGS_STR}")
@@ -589,7 +589,7 @@ file(GLOB IMAGE_IO_SRCS_LIST "${THIRD_PA
 set(PRIVATE_INCLUDE_DIR ${SOURCE_DIR}/include/ ${JPEG_INCLUDE_DIRS})
 set(PRIVATE_LINK_LIBS ${JPEG_LIBRARIES} Threads::Threads)
 if(UHDR_ENABLE_SMPTE2094_50)
-  list(APPEND PRIVATE_LINK_LIBS smpte2094_50_utils)
+  list(APPEND PRIVATE_LINK_LIBS smpte2094_50_utils absl_statusor absl_status)
 endif()
 if(UHDR_ENABLE_GLES)
   list(APPEND PRIVATE_INCLUDE_DIR ${EGL_INCLUDE_DIRS} ${OPENGLES3_INCLUDE_DIRS})
